Remove unused CoreIPC endpoints

sheeparegreat opened the pull request:
#### a610dff536527a055e005b137d102ad9fcade851
<pre>
Remove unused CoreIPC endpoints
<a href="https://bugs.webkit.org/show_bug.cgi?id=253729">https://bugs.webkit.org/show_bug.cgi?id=253729</a>
rdar://106568397

Reviewed by Alex Christensen.

This change removes ENABLE_UI_PROCESS_PDF_HUD and the following messages:
- Messages::WebPageProxy::SavePDFToFileInDownloadsFolder
- Messages::WebPageProxy::SavePDFToTemporaryFolderAndOpenWithNativeApplication
- Messages::WebPageProxy::OpenPDFFromTemporaryFolderWithNativeApplication

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewTestingMac.mm:
(-[WKWebView _pdfHUDs]):
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.h:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didCommitLoadForFrame):
(WebKit::WebPageProxy::resetStateAfterProcessTermination):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
(WebKit::WebPageProxy::openPDFFromTemporaryFolderWithNativeApplication): Deleted.
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::windowDidChangeBackingProperties):
(WebKit::WebViewImpl::mouseMoved):
(WebKit::WebViewImpl::mouseDown):
(WebKit::WebViewImpl::mouseUp):
* Source/WebKit/WebProcess/Plugins/PDF/PDFLayerControllerSPI.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate openWithNativeApplication]):
(-[WKPDFLayerControllerDelegate saveToPDF]):
(WebKit::PDFPlugin::PDFPlugin):
(WebKit::PDFPlugin::~PDFPlugin):
(WebKit::PDFPlugin::calculateSizes):
(WebKit::PDFPlugin::visibilityDidChange):
(WebKit::PDFPlugin::saveToPDF): Deleted.
(WebKit::PDFPlugin::openWithNativeApplication): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::savePDFToFileInDownloadsFolder): Deleted.
(WebKit::WebPage::savePDFToTemporaryFolderAndOpenWithNativeApplication): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:

Canonical link: <a href="https://commits.webkit.org/262597@main">https://commits.webkit.org/262597@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c3b90153bf3892ed22c36d8d4bada29b7e9e92aa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1868 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1901 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1962 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2794 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1966 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1845 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1975 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1950 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1738 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1887 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1690 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1680 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2639 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1682 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1662 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1577 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1570 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1716 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1709 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2857 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/1792 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1734 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1543 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/1927 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1659 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1673 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/436 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1816 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/1973 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/221 "Built successfully and passed tests") | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/550 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (changes requested):
This should delete the definition of ENABLE_UI_PROCESS_PDF_HUD and absolutely every instance of ENABLE(UI_PROCESS_PDF_HUD), removing the code where it is disabled.

achristensen07 commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
```diff
@@ -26,8 +26,6 @@
 #import "config.h"
 #import "WKPDFHUDView.h"
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.h:
```diff
@@ -23,8 +23,6 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/PageClient.h:
```diff
@@ -268,12 +268,10 @@ class PageClient : public CanMakeWeakPtr<PageClient> {
     virtual void didFailProvisionalLoadForMainFrame() { };
     virtual void didCommitLoadForMainFrame(const String& mimeType, bool useCustomContentProvider) = 0;
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -5270,9 +5270,7 @@ void WebPageProxy::didCommitLoadForFrame(FrameIdentifier frameID, FrameInfoData&
         requestPointerUnlock();
 #endif
         pageClient().setMouseEventPolicy(mouseEventPolicy);
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -8280,9 +8278,7 @@ void WebPageProxy::resetStateAfterProcessTermination(ProcessTerminationReason re
 
     resetStateAfterProcessExited(reason);
     stopAllURLSchemeTasks(m_process.ptr());
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -2042,15 +2039,13 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>
     bool canUseCredentialStorage() { return m_canUseCredentialStorage; }
     void setCanUseCredentialStorage(bool);
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.messages.in:
```diff
@@ -555,11 +549,9 @@ messages -> WebPageProxy {
     SpeechSynthesisResetState()
 #endif
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
```diff
@@ -647,9 +641,8 @@ static void getAllScriptsInPDFDocument(CGPDFDocumentRef pdfDocument, Vector<Reta
     auto& document = element.document();
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
     [m_pdfLayerController setDisplaysPDFHUDController:NO];
```
Could you actually add a respondsToSelector check for this?  That will make it easier for PDFKit to remove old code.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.h:
```diff
@@ -416,15 +416,13 @@ class WebPage : public API::ObjectImpl<API::Object::Type::BundlePage>, public IP
 
     void centerSelectionInVisibleArea();
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.h:
```diff
@@ -2117,9 +2110,7 @@ class WebPage : public API::ObjectImpl<API::Object::Type::BundlePage>, public IP
 
     WebCore::Color m_underlayColor;
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
```diff
@@ -623,12 +623,10 @@ GenerateSyntheticEditingCommand(enum:uint8_t WebKit::SyntheticEditingCommandType
 
     SetDefersLoading(bool defersLoading)
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
```
ENABLE(PDFKIT_PLUGIN)

sheeparegreat commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
```diff
@@ -191,7 +191,7 @@ static void createHostViewForExtensionIdentifier(void(^callback)(id hostViewCont
 #endif
 
-#if ENABLE(UI_PROCESS_PDF_HUD)
+#if ENABLE(PDFKIT_PLUGIN)
```
@achristensen07 This one too? (this change enabled iOS to build)

achristensen07 replied:
yes

sheeparegreat committed:
Remove unused CoreIPC endpoints
https://bugs.webkit.org/show_bug.cgi?id=253729
rdar://106568397

Reviewed by Alex Christensen.

This change removes ENABLE_UI_PROCESS_PDF_HUD and the following messages:
- Messages::WebPageProxy::SavePDFToFileInDownloadsFolder
- Messages::WebPageProxy::SavePDFToTemporaryFolderAndOpenWithNativeApplication
- Messages::WebPageProxy::OpenPDFFromTemporaryFolderWithNativeApplication

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewTestingMac.mm:
(-[WKWebView _pdfHUDs]):
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.h:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didCommitLoadForFrame):
(WebKit::WebPageProxy::resetStateAfterProcessTermination):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
(WebKit::WebPageProxy::openPDFFromTemporaryFolderWithNativeApplication): Deleted.
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::windowDidChangeBackingProperties):
(WebKit::WebViewImpl::mouseMoved):
(WebKit::WebViewImpl::mouseDown):
(WebKit::WebViewImpl::mouseUp):
* Source/WebKit/WebProcess/Plugins/PDF/PDFLayerControllerSPI.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate openWithNativeApplication]):
(-[WKPDFLayerControllerDelegate saveToPDF]):
(WebKit::PDFPlugin::PDFPlugin):
(WebKit::PDFPlugin::~PDFPlugin):
(WebKit::PDFPlugin::calculateSizes):
(WebKit::PDFPlugin::visibilityDidChange):
(WebKit::PDFPlugin::saveToPDF): Deleted.
(WebKit::PDFPlugin::openWithNativeApplication): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::savePDFToFileInDownloadsFolder): Deleted.
(WebKit::WebPage::savePDFToTemporaryFolderAndOpenWithNativeApplication): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:

Canonical link: https://commits.webkit.org/262597@main
